cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -604,16 +605,22 @@ externalSetupMethod verbosity options pkg bt mkargs = do
           -- With 'useDependenciesExclusive' we enforce the deps specified,
           -- so only the given ones can be used. Otherwise we allow the use
           -- of packages in the ambient environment, and add on a dep on the
-          -- Cabal library.
+          -- Cabal library (unless 'useDependencies' already contains one).
           --
           -- With 'useVersionMacros' we use a version CPP macros .h file.
           --
           -- Both of these options should be enabled for packages that have
           -- opted-in and declared a custom-settup stanza.
           --
```

Comment:
When we specify Cabal as a default setup dependency we want to prevent setup wrapper from adding another Cabal dependency. But we don't want it to set `-hide-all-packages`, since legacy setup scripts sometimes import stuff from libraries other than `base`/`Cabal`.